libraries/AP_NavEKF2/AP_NavEKF2_Measurements.cpp

```diff
@@ -38,7 +39,7 @@ void NavEKF2_core::readRangeFinder(void)
         // use data from two range finders if available
 
         for (uint8_t sensorIndex = 0; sensorIndex <= 1; sensorIndex++) {
-            if (frontend->_rng.status(sensorIndex) == RangeFinder::RangeFinder_Good) {
+            if ((frontend->_rng.get_orientation(sensorIndex) == ROTATION_PITCH_270) && (frontend->_rng.status(sensorIndex) == RangeFinder::RangeFinder_Good)) {
```

Comment:
Thanks for finding this.

I think it could happen before as well if 4 range finders were setup but only the 3rd one was healthy and thus it became the primary.  Perhaps we've never hit the problem because unless the user compiles their own code, we only ever support 2 range finders in our vehicles.

I'll talk to Paul about a solution..